bundles/framework/coordinatetool/plugin/CoordinatePluginHandler.js

```diff
@@ -0,0 +1,625 @@
+import { StateHandler, controllerMixin } from 'oskari-ui/util';
+import { showCoordinatePopup } from './CoordinatePopup';
+import { PLACEMENTS } from 'oskari-ui/components/window';
+
+const cloneJSON = (original) => JSON.parse(JSON.stringify(original));
+
+class UIHandler extends StateHandler {
+    constructor (plugin, mapModule, config, instance) {
+        super();
+        this.instance = instance;
+        this.plugin = plugin;
+        this.sandbox = Oskari.getSandbox();
+        this.loc = Oskari.getMsg.bind(null, 'coordinatetool');
+        this.mapModule = mapModule;
+        this.config = config;
+        this.setState({
+            xy: {},
+            displayXy: {},
+            loading: false,
+            popupControls: null,
+            showMouseCoordinates: false,
+            selectedProjection: this.mapModule.getProjection(),
+            originalProjection: this.mapModule.getProjection(),
+            emergencyInfo: null,
+            showReverseGeoCode: false
+        });
+        this.popupControls = null;
+        this.eventHandlers = this.createEventHandlers();
+        this.coordinateTransformationExtension =
+            Oskari.clazz.create('Oskari.mapframework.bundle.coordinatetool.plugin.CoordinateTransformationExtension',
```

Comment:
We probably don't need CoordinateTransformationExtension as is anymore. Maybe pick out some functions that are still used from that and add them to a helper.js that is imported with ES imports or something like that.